### PR TITLE
Add com.github.zocker_160.SyncThingy

### PR DESCRIPTION
**Pull request by zocker-160**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - `filesystem=host` makes sense I think, because sync targets can be anywhere
  - `share=network` is needed for syncing files over the network
  - `talk-name=org.kde.StatusNotifierWatcher` is needed to make the tray icon work
  - `talk-name=org.freedesktop.Notifications` is needed to be able to send notifications (service started and stopped for example)
  - `talk-name=org.freedesktop.portal.Background` is needed because it is meant to be run in the background all the time
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://github.com/zocker-160/SyncThingy
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

### Additional note

I have not tested this on an ARM machine, but I do not see any reason, why that shouldn't work, everything is built from source.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by zocker-160** [comment]
bot, built com.github.zocker160.SyncThingy

**Comment by zocker-160** [comment]
bot, build com.github.zocker160.SyncThingy

File: com.github.zocker160.SyncThingy.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        - type: git
+          url: https://github.com/zocker-160/SyncThingy.git
+          tag: 0.1
+          commit: d75b7836a1be25ee32eb15fba516c54466db5e52
+          x-checker-data:
+            type: git
+            tag-pattern: ^([\d.]+)$
+            
+        - type: file
+          url: https://raw.githubusercontent.com/syncthing/syncthing/main/assets/logo-512.png
```
[the remark this answers is not in the document]
Comment (hfiguiere) [reply com.github.zocker160.SyncThingy.yml:47]: actually insead of doing that here, I'd install it from the other module. (the one that is currently in another manifest file)

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: com.github.zocker160.SyncThingy.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    - --filesystem=host
+    - --talk-name=org.kde.StatusNotifierWatcher
+    - --talk-name=org.freedesktop.Notifications
+    - --talk-name=org.freedesktop.portal.Background
+
+modules:
+    - syncthing.yml
+
+    - name: SyncThingy
+      buildsystem: simple
```
[the remark this answers is not in the document]
Comment (zocker-160) [reply com.github.zocker160.SyncThingy.yml:27]: it totally does work, thanks for pointing that out

File: com.github.zocker160.SyncThingy.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (zocker-160) [reply com.github.zocker160.SyncThingy.yml:47]: understood, will do

File: com.github.zocker160.SyncThingy.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+app-id: com.github.zocker160.SyncThingy
```
[the remark this answers is not in the document]
Comment (zocker-160) [reply com.github.zocker160.SyncThingy.yml]: updated app ID in manifest and updated the manifest file name as well

File: syncthing.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+name: syncthing
```
[the remark this answers is not in the document]
Comment (zocker-160) [reply syncthing.yml]: you are right, I merged both into one

**Comment by zocker-160** [comment]
bot, build com.github.zocker_160.SyncThingy

**Comment by zocker-160** [comment]
bot, build

File: com.github.zocker_160.SyncThingy.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+        - type: file
+          url: https://raw.githubusercontent.com/syncthing/syncthing/9cc49aea7752a40f63011277bd1db4f85763b391/assets/logo-512.png
+          sha256: d0275a171c811209fa5d39b5ccdf7504fc356e40144cb98cb1d85e2acd4793e7
+              
+    - name: SyncThingy
+      buildsystem: cmake-ninja
+      post-install:
+        - install -D -m644 linux_packaging/SyncThingy.desktop -t /app/share/applications
+        - install -D -m644 linux_packaging/metainfo.xml -t /app/share/metainfo
```
Comment (hfiguiere) [inline com.github.zocker_160.SyncThingy.yml:57]: this clearly won't be found as it needs a `rename-appdata-file` or be installed with the right name

Comment (zocker-160) [reply com.github.zocker_160.SyncThingy.yml:57]: yes....I am so stupid

File: com.github.zocker_160.SyncThingy.yml [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+finish-args:
+    - --share=ipc
+    - --share=network
+    - --socket=fallback-x11
+    - --socket=wayland
+    - --device=dri
+    - --filesystem=host
+    - --talk-name=org.kde.StatusNotifierWatcher
+    - --talk-name=org.freedesktop.Notifications
+    - --talk-name=org.freedesktop.portal.Background
```
Comment (barthalion) [inline com.github.zocker_160.SyncThingy.yml:22]: But do you actually access this D-Bus name somehow? I glanced over the code and I can't find it. You don't need it to just let it run in the background.

Comment (zocker-160) [reply com.github.zocker_160.SyncThingy.yml]: I opted to comment it out, is that acceptable?

**Comment by zocker-160** [comment]
bot, build

File: com.github.zocker_160.SyncThingy.yml [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+finish-args:
+    - --share=ipc
+    - --share=network
+    - --socket=fallback-x11
+    - --socket=wayland
+    - --device=dri
+    - --filesystem=host
+    - --talk-name=org.kde.StatusNotifierWatcher
+    - --talk-name=org.freedesktop.Notifications
+    #- --talk-name=org.freedesktop.portal.Background # planned to be added later when actually needed
```
Comment (hfiguiere) [inline com.github.zocker_160.SyncThingy.yml:22]:
```suggestion
```

**Pull request closed without merging**